provider/hcloud/hcloud_discover_test.go

```diff
@@ -0,0 +1,93 @@
+package hcloud_test
```

Comment:
All of these tests could probably be consolidated into one table driven test, since all of them are testing pretty much the same thing (`Addr`), just with different parameters.

Reply:
yep, agree. added table driven tests. thanks once again.